.github/workflows/static-analysis.yml

```diff
@@ -0,0 +1,58 @@
+name: "Static Analysis"
+
+on:
+  pull_request:
+    paths:
+      - "source/**/*.php"
+      - ".github/workflows/*.yml"
+  workflow_call:
+    inputs:
+      ref:
+        description: "The git ref to check"
+        type: string
+        required: true
+
+env:
+  PHP_VERSION: "8.2"
+  # TODO: change to "stable" once 1.20.0 is released
+  # DRIVER_VERSION: "stable"
+  DRIVER_VERSION: "mongodb/mongo-php-driver@master"
+
+jobs:
+  psalm:
+    name: "Psalm"
+    runs-on: "ubuntu-22.04"
+
+    steps:
+      - name: "Checkout"
+        uses: "actions/checkout@v4"
+        with:
+          ref: ${{ github.event_name == 'workflow_dispatch' && inputs.ref || github.ref }}
+
+      - name: "Get SHA hash of checked out ref"
+        if: ${{ github.event_name == 'workflow_dispatch' }}
+        run: |
+          echo CHECKED_OUT_SHA=$(git rev-parse HEAD) >> $GITHUB_ENV
+
+      - name: "Setup"
+        uses: "./.github/actions/setup"
```

Comment:
This file need to be imported in the project: https://github.com/mongodb/mongo-php-library/blob/v1.19/.github/actions/setup/action.yml

Reply:
It is part of this PR, fixed a few min ago